engine/src/main/java/org/terasology/persistence/serializers/EventSerializer.java

```diff
@@ -135,15 +135,13 @@ private Event deserializeOnto(Event targetEvent, EntityData.Event eventData, Eve
 
         Serializer eventSerializer = typeSerializationLibrary.getSerializerFor(eventMetadata);
         ByteString.Output fieldIds = ByteString.newOutput();
-        for (ReplicatedFieldMetadata field : eventMetadata.getFields()) {
-            if (field.isReplicated()) {
-                EntityData.Value serializedValue = ((ProtobufPersistedData) eventSerializer.serialize(field, event, serializationContext)).getValue();
-                if (serializedValue != null) {
-                    eventData.addFieldValue(serializedValue);
-                    fieldIds.write(field.getId());
-                }
+        eventMetadata.getFields().stream().filter(field -> field.isReplicated()).forEach(field -> {
+            EntityData.Value serializedValue = ((ProtobufPersistedData) eventSerializer.serialize(field, event, serializationContext)).getValue();
+            if (serializedValue != null) {
+                eventData.addFieldValue(serializedValue);
+                fieldIds.write(field.getId());
```

Comment:
This method is called very often. I'd go with the original version, because I believe that it could be faster and the Java8 method don't make it much easier to read.